ci: Prevent duplicate GitHub action workflow runs

uristdwarf opened the pull request:
This PR prevents running twice the Github Actions pipeline used for Ganesha and unit tests.
Now, the pipeline runs only for Pull Requests.

ralcolea commented on .github/workflows/run-unit-and-ganesha-tests.yml:
```diff
@@ -13,6 +13,8 @@ jobs:
 
         env:
             SAUNAFS_TEST_TIMEOUT_MULTIPLIER: 2
+        if: github.event_name != 'pull_request' ||
```
Why not just remove Ganesha and unit tests pipeline for pull requests and use the result of the pipeline performed for the branch?

The only doubt is how this works for PRs created from external contributors. If in this case, the pipeline works for the branch, that needs to be created anyway, I think should be enough.
What do you think @uristdwarf?

uristdwarf replied:
Because we might need to run outside PR's.
Alternatively, we could make it PR only and remove the branch (As the GH action is now slower than the Jenkins branch pipeline)

ralcolea replied:
I also liked the idea of making it PR only. If you agree, we can add it to this PR :)

uristdwarf replied:
Totally.

ralcolea replied:
@uristdwarf I pushed a commit enabling the pipeline only for Pull Requests.
Please, confirm if you agree with this proposal or share any suggestion you might have :)

uristdwarf replied:
LGTM

uristdwarf replied:
You could probably remove my initial commit though and/or squash it

ralcolea replied:
@uristdwarf I removed your initial commit and rebased the branch on top of dev.
I will assign reviewers to proceed with the PR.

Thank you very much for your time :)

ralcolea committed:
ci: Enable Ganesha pipeline only for PRs

Previously, Ganesha and unit tests Github Actions pipeline ran twice,
for branches and Pull Requests.

This commit enables Ganesha and unit tests pipeline to run only for
Pull Requests. It also removes the previous branches filter, because
pull_request event, by default runs the pipeline for any target branch.

Co-authored-by: Urmas Rist <urmas@leil.io>

Signed-off-by: Crash <crash@leil.io>

lgsilva3087 reviewed (approved):
LGTM

rolysr reviewed (approved):
LGTM.
